test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CustomCcmRule.java

```diff
@@ -37,7 +40,20 @@ public class CustomCcmRule extends BaseCcmRule {
   @Override
   protected void before() {
     if (CURRENT.get() == null && CURRENT.compareAndSet(null, this)) {
-      super.before();
+      try {
+        super.before();
+      } catch (Exception e) {
+        // ExternalResource will not call after() when before() throws an exception
+        // Let's try and clean up and release the lock we have in CURRENT
+        LOG.warn(
+            "Error in CustomCcmRule before() method, attempting to clean up leftover state", e);
+        try {
+          after();
```

Comment:
Another question, will it break `RuleChain` sequence ?  
Meaning if `CustomCcmRule` is part of the `RuleChain`:
`CustomCcmRule`, `Rule#1`
then `RuleChain.before` calls `CustomCcmRule.before` and then `Rule#1.before`
when test or any rule fails it is going to call:
`RuleChain.after` which is going to call `Rule#1.after` and only then `CustomCcmRule.after`.
If you do call `after()` here it changes the sequence and can cause unexpected consuquences.